packages/region/src/services/region-module.ts

```diff
@@ -0,0 +1,230 @@
+      {},
+      sharedContext
+    )
+
+    let currencyMap = new Map(currencies.map((c) => [c.code.toLowerCase(), c]))
+    for (const reg of data) {
+      const lowerCasedCurrency = reg.currency_code.toLowerCase()
+      if (!currencyMap.has(lowerCasedCurrency)) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Currency with code: ${reg.currency_code} was not found`
+        )
+      }
+
+      reg.currency = currencyMap.get(lowerCasedCurrency) as RegionCurrencyDTO
+    }
+
+    const result = await this.regionService_.create(data, sharedContext)
+
+    return result
+  }
+
+  async update(
+    data: UpdateRegionDTO[],
+    sharedContext?: Context
+  ): Promise<RegionDTO[]>
+  async update(
+    data: UpdateRegionDTO,
+    sharedContext?: Context
+  ): Promise<RegionDTO>
+  @InjectTransactionManager("baseRepository_")
+  async update(
+    data: UpdateRegionDTO | UpdateRegionDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<RegionDTO | RegionDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+
+    const result = await this.regionService_.update(input, sharedContext)
+
+    return await this.baseRepository_.serialize<RegionDTO[]>(
+      Array.isArray(data) ? result : result[0],
+      {
+        populate: true,
+      }
+    )
+  }
+
+  @InjectManager("baseRepository_")
+  private async createCountriesAndCurrencies(
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<void> {
+    await promiseAll([
+      await this.maybeCreateCountries(sharedContext),
+      await this.maybeCreateCurrencies(sharedContext),
+    ])
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  private async maybeCreateCountries(
+    @MedusaContext() sharedContext: Context
+  ): Promise<void> {
+    const [country] = await this.countryService_.list(
+      {},
+      { select: ["id"], take: 1 }
+    )
+
+    let countsToCreate: CreateCountryDTO[] = []
+    if (!country) {
+      countsToCreate = DefaultsUtils.defaultCountries.map((c) => ({
+        iso_2: c.alpha2.toLowerCase(),
+        iso_3: c.alpha3.toLowerCase(),
+        num_code: c.numeric,
+        name: c.name.toUpperCase(),
+        display_name: c.name,
+      }))
+    }
+
+    if (countsToCreate.length) {
+      await this.countryService_.create(countsToCreate, sharedContext)
+    }
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  private async maybeCreateCurrencies(
+    @MedusaContext() sharedContext: Context
+  ): Promise<void> {
+    const [currency] = await this.currencyService_.list(
+      {},
+      { select: ["id"], take: 1 }
+    )
+
+    let currsToCreate: CreateCurrencyDTO[] = []
+    if (!currency) {
+      currsToCreate = Object.entries(DefaultsUtils.defaultCurrencies).map(
+        ([code, currency]) => ({
+          code: code.toLowerCase(),
+          symbol: currency.symbol,
+          symbol_native: currency.symbol_native,
+          name: currency.name,
+        })
+      )
+    }
```

Comment:
suggestion(non-blocking): compare count in db with `defaultCurrencies` count:

```
const [, count] = await this.currencyService_.listAndCount({}, {select: ["id"], take: 1 })

if (count !== Object.entries(defaultCurrencies).length) {
  ...
}
```
This way we can add currencies by updating defaultCurrencies. 

Note: the creation would have to account for this somehow to ensure it's only the difference that is created.